clang/lib/3C/RewriteUtils.cpp

```diff
@@ -198,36 +195,69 @@ static void emit(Rewriter &R, ASTContext &C, std::string &OutputPostfix) {
         continue;
       }
 
-      // -output-postfix mode
-
       // Produce a path/file name for the rewritten source file.
-      // That path should be the same as the old one, with a
-      // suffix added between the file name and the extension.
-      // For example \foo\bar\a.c should become \foo\bar\a.checked.c
-      // if the OutputPostfix parameter is "checked" .
-      std::string PfName = sys::path::filename(FE->getName()).str();
-      std::string DirName = sys::path::parent_path(FE->getName()).str();
-      std::string FileName = sys::path::remove_leading_dotslash(PfName).str();
-      std::string Ext = sys::path::extension(FileName).str();
-      std::string Stem = sys::path::stem(FileName).str();
-      std::string NFile = Stem + "." + OutputPostfix + Ext;
-      if (!DirName.empty())
-        NFile = DirName + sys::path::get_separator().str() + NFile;
-
+      std::string NFile;
       std::error_code EC;
+
```

Comment:
```suggestion
      // We now know that we are using either OutputPostfix or OutputDir mode
      // because stdout mode is handled above. OutputPostfix defaults to "-"
      // when it's not provided, so any other value means that we should use
      // OutputPostfix. Otherwise, we must be in OutputDir mode. 
```